conf/i18n/translatehelpervisitor.js

```diff
@@ -0,0 +1,169 @@
+const Handlebars = require('handlebars');
+const TranslationPlaceholderFactory = require('./translationplaceholderfactory');
+
+/**
+ * TranslateHelperVisitor accepts a handlebars AST, and replaces all translate placeholders
+ * with either static translations, or a runtime translation helper.
+ */
+class TranslateHelperVisitor {
+  constructor (translationResolver) {
+    this._visitor = new Handlebars.Visitor();
+    // This line puts the Handlebars.Visitor instance into mutation mode.
+    // https://github.com/handlebars-lang/handlebars.js/blob/master/docs/compiler-api.md#ast-visitor.
+    this._visitor.mutating = true;
+    this._visitor.MustacheStatement = this._MustacheStatement.bind(this);
+
+    this._translationResolver = translationResolver;
+    this._validHelpers = ['translate'];
+    this._runtimeTranslationHelper = 'runtimeTranslation';
+    this._contextParam = 'context';
+    this._pluralFormParam = 'pluralForm';
+  }
+
+  /**
+   * Dispatches TranslateHelperVisitor on a Handlebars AST.
+   * @param {hbs.AST.Program} ast
+   */
+  accept (ast) {
+    this._visitor.accept(ast);
+  }
+
+  /**
+   * _MustacheStatement() is automatically called on any MustacheStatement nodes by the visitor.
+   * Returning undefined leaves the node unaffected, otherwise it replaces it with the
+   * returned value.
+   * @param {hbs.AST.MustacheStatement} statement
+   * @returns {hbs.AST.MustacheStatment|undefined}
+   */
+  _MustacheStatement (statement) {
+    const isTranslationHelper = this._validHelpers.includes(statement.path.original);
```

Comment:
Can we add a verb to this function name?

Reply:
sure I was trying to follow the method signature of Handlebars.Visitor but we don't have to be restrained by them